rmb-sdk-go/peer/peer.go

```diff
@@ -80,14 +81,28 @@ func getIdentity(keytype string, mnemonics string) (substrate.Identity, error) {
 // Call() will panic if called while the directClient's context is canceled.
 func NewPeer(
 	ctx context.Context,
-	keytype string,
 	mnemonics string,
-	relayURL string,
-	session string,
 	sub *substrate.Substrate,
-	enableEncryption bool,
-	handler Handler) (*Peer, error) {
-	identity, err := getIdentity(keytype, mnemonics)
+	handler Handler,
+	opts ...PeerOpt) (*Peer, error) {
+
+	cfg := &peerCfg{
+		relayURL:         "wss://relay.grid.tf",
+		session:          "",
+		enableEncryption: false,
+		keyType:          KeyTypeSr25519,
+		schema:           encoder.DefaultSchema,
+	}
+	for _, o := range opts {
+		o(cfg)
+	}
+
+	e, err := encoder.NewEncoder(cfg.schema)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is not the right way. This basically means that NewEncoder need to be modified everytime you have to implement a new encoding type. 

What if in another app i wanna use this with "binary" encoding, do i have to fork and modify your library before i can do this ?

The correct way to do this instead is that your option instead must be  `WithEncoder` (or something) that takes an instance of Encoder type. 

If encoder is not set (yourc `cfg.encoder` is still nil after applying all the Withs) then you can then create the default JsonEncoder and use that.